qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -801,7 +807,7 @@ def is_controlled(gate):
                     pi2_decomposer = TwoQubitBasisDecomposer(
                         pi2_basis,
                         euler_basis=basis_1q,
-                        basis_fidelity=basis_2q_fidelity,
+                        basis_fidelity=basis_2q_fidelity,  # TODO: doesn't accept dict?
```

Comment:
`TwoQubitBasisDecomposer` accepts float as `basis_fidelity`, but `basis_2q_fidelity` is dict?

Reply:
it seems there is some variable reuse later, for calling `XXDecomposer`. I would rename `basis_2q_fidelity` to `basis_2q_fidelities` after `basis_2q_fidelity = {}` (renaming to `basis_2q_fidelities = {}`).